packages/dev/core/src/Shaders/ShadersInclude/pbrBlockSubSurface.fx

```diff
@@ -243,58 +339,64 @@ struct subSurfaceOutParams
             float refractionLOD = getLodFromAlphaG(vRefractionMicrosurfaceInfos.x, refractionAlphaG);
         #endif
 
-        #ifdef LODBASEDMICROSFURACE
-            // Apply environment convolution scale/offset filter tuning parameters to the mipmap LOD selection
-            refractionLOD = refractionLOD * vRefractionMicrosurfaceInfos.y + vRefractionMicrosurfaceInfos.z;
-
-            #ifdef SS_LODINREFRACTIONALPHA
-                // Automatic LOD adjustment to ensure that the smoothness-based environment LOD selection
-                // is constrained to appropriate LOD levels in order to prevent aliasing.
-                // The environment map is first sampled without custom LOD selection to determine
-                // the hardware-selected LOD, and this is then used to constrain the final LOD selection
-                // so that excessive surface smoothness does not cause aliasing (e.g. on curved geometry
-                // where the normal is varying rapidly).
-
-                // Note: Shader Model 4.1 or higher can provide this directly via CalculateLevelOfDetail(), and
-                // manual calculation via derivatives is also possible, but for simplicity we use the 
-                // hardware LOD calculation with the alpha channel containing the LOD for each mipmap.
-                float automaticRefractionLOD = UNPACK_LOD(sampleRefraction(refractionSampler, refractionCoords).a);
-                float requestedRefractionLOD = max(automaticRefractionLOD, refractionLOD);
+        #ifdef SS_DISPERSION
+        float realIOR = 1.0 / ior;
+            float iorDispersionSpread = 0.02 * dispersion * (realIOR - 1.0);
+            vec3 iors = vec3(ior + iorDispersionSpread, ior, ior - iorDispersionSpread);
+            for (int i = 0; i < 3; i++) {

```

Comment:
I think we can do something like this to avoid duplicating the call to `sampleEnvironmentRefraction`:
```glsl
#ifdef SS_DISPERSION
    float realIOR = 1.0 / ior;
    float iorDispersionSpread = 0.02 * dispersion * (realIOR - 1.0);
    vec3 iors = vec3(ior + iorDispersionSpread, ior, ior - iorDispersionSpread);
    for (int i = 0; i < 3; i++) {
#else
    vec2 iors = vec2(ior);
    int i = 0; {
#endif
```

Reply:
Done